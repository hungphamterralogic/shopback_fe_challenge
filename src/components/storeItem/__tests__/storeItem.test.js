import React from 'react';

import ReactTestRenderer from 'react-test-renderer';

import detectDevice from 'utils/detectDevice';

import StoreItem from '../storeItem';

const id = 'idTest';
const item = {
  id,
  logoUrl: 'logoUrlTest',
  displayText: 'displayTextTest'
};

describe('StoreItem', () => {
  describe('display properly', () => {
    test('is mobile', () => {
      jest.spyOn(detectDevice, 'isMobile').mockImplementationOnce(() => true);

      const selectStore = () => {};
      const renderer = ReactTestRenderer.create(
        <StoreItem item={item} selectStore={selectStore} />
      );

      expect(renderer.toJSON()).toMatchSnapshot();
    });

    test('is not mobile', () => {
      jest.spyOn(detectDevice, 'isMobile').mockImplementationOnce(() => false);

      const selectStore = () => {};
      const renderer = ReactTestRenderer.create(
        <StoreItem item={item} selectStore={selectStore} />
      );

      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });

  test('on click', () => {
    const selectStore = jest.fn();
    const renderer = ReactTestRenderer.create(<StoreItem item={item} selectStore={selectStore} />);

    (function clickOnStoreItem() {
      const storeItem = renderer.root.find(el => el.type === 'div');
      storeItem.props.onClick();
    })();

    expect(selectStore.mock.calls.length).toBe(1);
    expect(selectStore).toHaveBeenCalledWith(id);
  });
});
