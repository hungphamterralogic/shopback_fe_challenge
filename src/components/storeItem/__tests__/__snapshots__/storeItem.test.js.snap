// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreItem display properly is mobile 1`] = `
<div
  className="StoreItem-storeItem-0-1-1 StoreItem-storeItemMobile-0-1-2"
  onClick={[Function]}
>
  <img
    alt="icon"
    className="StoreItem-icon-0-1-3"
    src="logoUrlTest"
  />
  <span
    className="StoreItem-name-0-1-4"
  >
    displayTextTest
  </span>
</div>
`;

exports[`StoreItem display properly is not mobile 1`] = `
<div
  className="StoreItem-storeItem-0-1-1"
  onClick={[Function]}
>
  <img
    alt="icon"
    className="StoreItem-icon-0-1-3"
    src="logoUrlTest"
  />
  <span
    className="StoreItem-name-0-1-4"
  >
    displayTextTest
  </span>
</div>
`;
